src/Microsoft.AspNetCore.Mvc.Core/Internal/ControllerActionInvoker.cs

```diff
@@ -672,12 +690,15 @@ private Task Next(ref State next, ref Scope scope, ref object state, ref bool is
 
                         if (exceptionContext != null)
                         {
-                            if (exceptionContext.Result != null && !exceptionContext.ExceptionHandled)
+                            if (exceptionContext.Result != null ||
+                                exceptionContext.Exception == null ||
+                                exceptionContext.ExceptionHandled)
```

Comment:
So, no change needed here. I'd like to discuss suppressing the `Exception` in the "`Result`-only" case for 2.0.